workflow/lib/shared/eval_segmentation.py

```diff
@@ -88,3 +94,274 @@ def plot_cell_density_heatmap(df_cells, shape="square", plate="6W", **kwargs):
+        default_cell_diameter (float, optional): Reference cell diameter for comparison.
+            If None, only optimal parameters are shown. Defaults to None.
+        default_nuclei_diameter (float, optional): Reference nuclei diameter.
+            Required if default_cell_diameter is provided. Defaults to None.
+        default_cellprob_threshold (float, optional): Reference cell probability threshold.
+            Required if default_cell_diameter is provided. Defaults to None.
+        default_flow_threshold (float, optional): Reference flow threshold.
+            Required if default_cell_diameter is provided. Defaults to None.
+        channel_cmaps (list of str, optional): List of colormap names for each channel.
+            Only used when segmentation_process is "phenotype_process".
+        prepare_cellpose_kwargs (dict, optional): Keywords for prepare_cellpose function.
+            Only used when segmentation_process is "sbs_process".
+
+    Returns:
+        pandas.DataFrame: Statistics grouped by parameter combinations, containing columns:
+            initial_nuclei_mean, initial_cells_mean, final_cells_mean, final_nuclei_mean,
+            cell_retention_mean, nuclei_retention_mean, measurement_count, combined_score
+        str: Formatted summary text containing performance metrics for optimal and default parameters
+        Micropanel: Visualization comparing optimal and default segmentation results (if defaults provided)
+    """
+
+    def paths_segmentation_paramsearch(base_path):
+        """Generate paths for nuclei, cell, and aligned image files from a stats path.
+
+        Takes a segmentation stats file path and converts it to corresponding paths
+        for the nuclei, cell mask, and illumination-corrected image files.
+
+        Args:
+            base_path (str): Path to segmentation stats TSV file, typically containing
+                '/tsvs/' and ending with '_segmentation_stats.tsv'
+
+        Returns:
+            str: Path to nuclei mask TIFF file (_nuclei.tiff)
+            str: Path to cell mask TIFF file (_cells.tiff)
+            str: Path to illumination-corrected TIFF file (__illumination_corrected.tiff)
+        """
+        # Convert Path to string if needed
+        base_path = str(base_path)
+
+        # Replace /tsvs/ with /images/
+        base_path = base_path.replace("/tsvs/", "/images/")
+
+        # Remove extension "_segmentation_stats.tsv"
+        prefix = base_path.replace("_segmentation_stats.tsv", "")
+
+        # Create paths for nuclei and cells
+        nuclei_path = f"{prefix}_nuclei.tiff"
+        cells_path = f"{prefix}_cells.tiff"
+
+        # Get aligned image path by replacing /paramsearch/tsvs/ with /
+        aligned_path = base_path.replace("/paramsearch/", "/")
+        aligned_path = (
+            aligned_path.split("__paramsearch")[0] + "__illumination_corrected.tiff"
+        )
+
+        return nuclei_path, cells_path, aligned_path
```

Comment:
I've run into issues with code like this before (hard coding names for paths, metadata, etc). I don't think we should worry about it too much right now but if we are consistently messing with path names we should add a standardized way to do stuff like this so we/other devs do the same thing across Brieflow.

Reply:
Agreed. Maybe something to open an issue about? But for now, need to get it to work.